service/demo/DP_SQL_demo.ipynb

```diff
@@ -0,0 +1,286 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## SQL Queries with Differentially Private Set Union (DPSU)\n",
+    "\n",
+    "In the context of the WhiteNoise SQL scenario, consider a guiding example: there exists a dataset with Reddit users and their posts, keeping in mind that a single user could have multiple posts. An analyst would like to release a report containing the counts of the number of bigrams per user while preserving privacy. We'd like to increase the representation of bigrams contained in this report without violating privacy bounds.\n",
+    "\n",
+    "You can imagine that it might be easy to identify an author's bigrams if she has only posted a few times and therefore has very few bigrams. To resolve this threat, previously, we would have to drop bigrams whose noisy counts were below a certain threshold. However, this process was wasteful because we allocated privacy budget to all rows uniformly even if their counts already exceeded the threshold.\n",
+    "\n",
+    "With DPSU, we resolve this issue by adding noise in a dependent fashion. This increases the dimensionality of the final dataset, and we therefore have less data loss and get a richer representation of bigrams. \n",
+    "\n",
+    "The code below showcases DPSU support with pandas."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import re\n",
+    "import os\n",
+    "import subprocess\n",
+    "import pandas as pd\n",
+    "from opendp.whitenoise.metadata import CollectionMetadata\n",
+    "from opendp.whitenoise.metadata.collection import Table, String\n",
+    "from opendp.whitenoise.sql import PrivateReader, PandasReader"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def find_ngrams(input_list, n):\n",
+    "    return input_list if n == 1 else list(zip(*[input_list[i:] for i in range(n)]))\n",
+    "\n",
+    "def _download_file(url, local_file):\n",
+    "    try:\n",
+    "        from urllib import urlretrieve\n",
+    "    except ImportError:\n",
+    "        from urllib.request import urlretrieve\n",
+    "    urlretrieve(url, local_file)\n",
+    "\n",
+    "root_url = subprocess.check_output(\"git rev-parse --show-toplevel\".split(\" \")).decode(\"utf-8\").strip()\n",
```

Comment:
We should remove this, it should no longer be needed. You can make root_url -> "."  The current directory